content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry/index.md

```diff
@@ -0,0 +1,294 @@
+---
+title: Collecting file-based Java logs with OpenTelemetry
+linkTitle: Collecting file-based Java logs with OpenTelemetry
+date:
+author: >
+  [Cyrille Le Clerc](https://github.com/cyrille-leclerc) (Grafana Labs), [Gregor
+  Zeitlinger](https://github.com/zeitlinger) (Grafana Labs)
+issue: https://github.com/open-telemetry/opentelemetry.io/issues/5606
+sig: Java, Specification
+# prettier-ignore
+cSpell:ignore: Clerc cust Cyrille Dotel Gregor Logback logback otlphttp otlpjson resourcedetection SLF4J stdout Zeitlinger
+---
+
+If you want to get logs from your Java application ingested into an
+OpenTelemetry-compatible logs backend, the easiest and recommended way is using
+an OpenTelemetry protocol (OTLP) exporter. However, some scenarios require logs
+to be output to files or stdout due to organizational or reliability needs.
+
+A common approach to centralize logs is to use unstructured logs, parse them
+with regular expressions, and add contextual attributes.
+
+However, regular expression parsing is problematic. They become complex and
+fragile quickly when handling all log fields, line breaks in exceptions, and
+unexpected log format changes. Parsing errors are inevitable with this method.
+
+## Solution
+
+The OpenTelemetry Java Instrumentation agent and SDK now offer an easy solution
+to convert logs from frameworks like SLF4J/Logback or Log4j2 into OTel-compliant
+JSON logs on stdout with all resource and log attributes.
+
+This is a true turnkey solution:
+
+- No code or dependency changes, just a few configuration adjustments typical
+  for production deployment.
+- No complex field mapping in the log collector. Just use the
+  [OTLP/JSON connector](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/connector/otlpjsonconnector)
+  to ingest the payload.
+- Automatic correlation between logs, traces, and metrics.
+
+This blog post shows how to set up this solution step by step.
+
+- In the first part, we'll show how to configure the Java application to output
+  logs in the OTLP/JSON format.
+- In the second part, we'll show how to configure the OpenTelemetry Collector to
+  ingest the logs.
+- Finally, we'll show a Kubernetes-specific setup to handle container logs.
+
+## Reference architecture
+
+The deployment architecture looks like the following:
+
+![OTLP/JSON Architecture](otlpjson-architecture.png)
```

Comment:
This diagram is really busy, and the content too small. I'd suggest the following improvements:

- Remove the following text from the diagram and add it instead as prose to this section (before or after the image):
  - "Logs contextualized with OTel metadata including: ..."
  - "experimental-otlp/stout" exporter Available in Java Agent v2.10"
- Move the "Observability Platform" block to be below the rest so that the image becomes less wide -- that is, favor vertical image real estate rather than horizontal. Or, if you prefer, put the app at the top of the diagram with the OTel components below. Something (very roughly) like what I show below. 

> <img width="772" alt="image" src="https://github.com/user-attachments/assets/e46b347f-f800-4d6e-9146-2138880b0fde">

Reply:
What do you think of this style?
![image](https://github.com/user-attachments/assets/00f1904b-3c0f-4db0-b3ad-ccf26f19dbe4)

Reply:
Here is a proposal of half horizontal and half vertical style.

@svrnm @zeitlinger do you prefer this or the horizontal style?

![image](https://github.com/user-attachments/assets/9478454d-51aa-4605-ac4d-790ad7d7c1d2)

Reply:
I like the  new one better

Reply:
lgtm

@chalin wdyt?

Reply:
I pushed the new "1/2 horizontal & 1/2 vertical style", LGTM, thanks for he suggestion!